frontend/src/forms/subforms/AddressForm.tsx

```diff
@@ -41,6 +41,20 @@ const AddressForm = <T extends any>(props: AddressProps & FormikProps<T>) => {
     return nameSpace ? `${nameSpace}.${fieldName}` : fieldName;
   };
 
+  const postalCodeFormatter = (postal: string) => {
```

Comment:
I think this may be overthought.  All you really want to do is add a space at the 3 index position and uppercase.  You're not validating in this function as far as I can tell.  If you can find a nice non-mutating solution that inserts a space at the index position 3.
```js
var postal = 'a8a8a8'.split('');
postal.splice(3,0,' '); // mutating however
var formatted = postal.join('').toUpperCase();
```

Reply:
my recommendation would be to use capturing groups to get the first three and last three characters, and then format those as desired. That way you can use a single regex and you don't need any additional processing functions outside of the regex.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match

Reply:
Gotchya, thanks. Adjusting now